backend/prisma/schema.prisma

```diff
@@ -11,71 +11,118 @@ datasource db {
   relationMode = "prisma"
 }
 
-model School {
-  id       Int       @id @default(autoincrement())
-  name     String
-  domain   String
-  students Student[]
-  users    User[]
-
-  @@map("school")
+//
+//
+// ENUMS
+//
+//
+// FOR USE IN USER, ORGANIZATION, EVENT
+
+// FOR USE IN PARTICIPATION
+enum ParticipationStatus {
+  Going
+  Interested
+  NotInterested
 }
 
-model Student {
-  id       Int     @id @default(autoincrement())
-  school   School  @relation(fields: [schoolID], references: [id])
-  schoolID Int
-  email    String
-  username String
-  name     String
-  password String
-  otp      String
-  jwt      String
-  status   Boolean
+enum ProgramType {
+  Bachelors
+  Masters
+}
 
-  @@index([schoolID])
-  @@map("student")
+enum EventStatus {
+  Verified
+  NonVerified
+}
+
+//
+//
+// CORE MODELS
+//
+//
+model Institution {
+  id     String @id @default(uuid())
+  name   String
+  domain String
+
+  user User[]
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  @@map("institution")
 }
 
 model User {
-  id                Int                    @id @default(autoincrement())
-  username          String                 @unique
-  firstName         String                 @map("first_name")
-  lastName          String                 @map("last_name")
-  email             String                 @unique
-  password          String                 @map("password")
-  yearOfStudy       Int                    @map("year_of_study")
-  schoolId          Int                    @map("school_id")
-  isVerified        Boolean                @default(false) @map("is_verified")
-  profilePic        String?                @unique @map("profile_pic")
-  school            School                 @relation(fields: [schoolId], references: [id], onDelete: Cascade)
+  id String @id @default(uuid())
+
+  username  String @unique
+  firstName String
+  lastName  String
+
+  email    String @unique
+  password String
+
+  otp        String? @default("")
+  jwt        String? @default("")
+  isVerified Boolean @default(false)
+
+  profilePic String? @unique
+
+  institution   Institution @relation(fields: [institutionId], references: [id], onDelete: Cascade)
+  institutionId String
+
+  enrollments Enrollment[]
+
+  posts    Post[]
+  comments Comment[]
+
   events            Event[]
   UserEventResponse UserEventResponse[]
-  posts             Post[]
-  comments          Comment[]
-  organizationRoles UserOrganizationRole[]
-  enrollments       Enrollment[]
-  subscriptions     TopicSubscription[]
 
-  otp    String
-  jwt    String
-  status Boolean
+  subscriptions        TopicSubscription[]
+  UserOrganizationRole UserOrganizationRole[]
 
-  @@index([schoolId])
+  @@index([email])
+  @@index([institutionId])
   @@map("user")
 }
 
+model Enrollment {
```

Comment:
Enrollment doesn't need an id. We can just use a combination of userId and programId
also needs snake case mapping
Why is degreeType is deleted?